Submitted By:            Randy McMurchy <randy_at_linuxfromscratch_dot_org>
Date:                    2004-10-31
Initial Package Version: 2.4.1
Upstream Status:         Not submitted (too much of a hack)
Origin:                  http://lists.gnu.org/archive/html/bug-gnu-utils/2003-01/msg00207.html
Description:             Fixes C++ error using GCC >= 3.x.x
                         Note: This patch is not meant to be used directly. It
                         is called by libplotter/Makefile which is created
                         with a patched Makefile.in. You'll also need:
                         plotutils-2.4.1-libplotter_Makefile-1.patch
                         plotutils-2.4.1-n_write_cc_fix-1.patch
                         plotutils-2.4.1-z_write_cc_fix-1.patch
                         plotutils-2.4.1-g_write_cc_fix-1.patch

diff -Naur ../../plotutils-2.4.1-orig/libplot/i_rle.c ./i_rle.cc
--- ../../plotutils-2.4.1-orig/libplot/i_rle.c  1999-06-27 16:58:10.000000000 +0000
+++ ./i_rle.cc  2004-10-31 18:33:29.357017888 +0000
@@ -78,7 +78,7 @@
   else if (rle->outstream)
     {
       rle->outstream->put ((unsigned char)(rle->oblen));
-      rle->outstream->write (&(rle->oblock[0]), rle->oblen);
+      rle->outstream->write (reinterpret_cast<const char*>(&(rle->oblock[0])), rle->oblen);
     }
 #endif

